ReactiveArchitecture/Classes/core/extension/Observable+Optional.swift

```diff
@@ -0,0 +1,30 @@
+//
+//  Observable+Optional.swift
+//  ReactiveArchitecture
+//
+//  Created by Maximilian Clarke on 22/1/18.
+//  Copyright © 2018 leonardis. All rights reserved.
+//
+
+import Foundation
+import RxSwift
+
+// From https://gist.github.com/alskipp/e71f014c8f8a9aa12b8d8f8053b67d72
+
+protocol OptionalType {
+    associatedtype Wrapped
+    
+    var optional: Wrapped? { get }
+}
+
+extension Optional: OptionalType {
+    public var optional: Wrapped? { return self }
+}
+
+extension Observable where Element: OptionalType {
+    func ignoreNil() -> Observable<Element.Wrapped> {
```

Comment:
Added this for easier strong type mapping to subclasses during an observable flow. Optional map to class -> ignore nils.

Reply:
Nice...not 100% sure what it does but the less manual casting the better

Reply:
FYI, RxSwiftExt (a collection of RxSwift extensions from the RxSwift contributors) has an option similar to this already called `.unwrap()`